crates/iota-json-rpc/src/governance_api.rs

```diff
@@ -773,7 +773,18 @@ fn backfill_rates(
     // ensure epochs are processed in increasing order
     rates.sort_unstable_by_key(|(epoch_id, _)| *epoch_id);
 
-    let mut filled_rates: Vec<(EpochId, PoolTokenExchangeRate)> = Vec::with_capacity(rates.len());
+    // Check if there are any gaps in the epochs
+    let min_epoch = rates.first().unwrap().0;
+    let max_epoch = rates.last().unwrap().0;
+    let expected_len = (max_epoch - min_epoch + 1) as usize;
+
+    // Only perform backfilling if there are gaps
+    if rates.len() == expected_len {
+        rates.reverse();
+        return rates;
+    }
+
+    let mut filled_rates: Vec<(EpochId, PoolTokenExchangeRate)> = Vec::with_capacity(expected_len);
     let mut missing_rates = Vec::new();
```

Comment:
We already know how many rates are missing at this point. We could initialise with a capacity already.

Reply:
Thanks. Missed this. Added at [86467c0](https://github.com/iotaledger/iota/pull/5184/commits/86467c034510d51e2d2423220ff7e4d2000c9839).